test/uk/gov/hmrc/transitmovementsauditing/controllers/AuditControllerSpec.scala

```diff
@@ -97,6 +99,35 @@ class AuditControllerSpec
   )
   private val invalidJsonDetailsStream = Source.single(ByteString("""{ "data": {"path": "some-path"}, "payload": { "test": "123" }} """.mkString))
 
+  private val someValidDetailsJson =
+    """
+      |{
+      |    "metadata": {
+      |        "path": "some-path"
+      |    },
+      |    "payload": {
+      |            "test": "123"
+      |    }
+      |}
+      |""".stripMargin
+
+  private val someValidFullDetailsJson =
+    """
+      |{
+      |    "metadata": {
+      |        "path": "some-path",
+      |        "movementId": "movementId",
+      |        "messageId": "messageId",
+      |        "enrolmentEORI": "enrolmentEORI",
+      |        "movementType": "departure",
+      |        "messageType": "IE015"
+      |    },
+      |    "payload": {
+      |            "test": "123"
+      |    }
+      |}
+      |""".stripMargin
```

Comment:
Similar thing here. All you're ever doing is `Json.parse(someValidDetailsJson).validate[Details].get` in these cases. Why not just have the `Details` objects -- and in fact, better yet, use property based testing?

Reply:
yes, used the Details obj now